flocker/provision/_install.py

```diff
@@ -156,89 +155,154 @@ def task_install_kernel_devel():
+def task_enable_docker(distribution):
     """
     Start docker and configure it to start automatically.
     """
+    if distribution in ('fedora-20', 'centos-7'):
+        return sequence([
+            run_from_args(["systemctl", "enable", "docker.service"]),
+            run_from_args(["systemctl", "start", "docker.service"]),
+        ])
+    elif distribution == 'ubuntu-14.04':
+        # Ubuntu enables docker service during installation
+        return sequence([])
+    else:
+        raise NotImplementedError()
+
+
+def open_firewalld(service):
+    """
+    Open firewalld port for a service.
+
+    :param str service: Name of service.
+    """
     return sequence([
-        run_from_args(["systemctl", "enable", "docker.service"]),
-        run_from_args(["systemctl", "start", "docker.service"]),
-    ])
+        run_from_args(command + [service])
+        for command in [['firewall-cmd', '--permanent', '--add-service'],
+                        ['firewall-cmd', '--add-service']]])
 
 
-def configure_firewalld(rule):
+def open_ufw(service):
     """
-    Configure firewalld with a given rule.
+    Open ufw port for a service.
 
-    :param list rule: List of `firewall-cmd` arguments.
+    :param str service: Name of service.
     """
     return sequence([
-        run_from_args(command + rule)
-        for command in [['firewall-cmd', '--permanent'],
-                        ['firewall-cmd']]])
+        run_from_args(['ufw', 'allow', service])
+        ])
 
 
-def task_enable_flocker_control():
+def task_enable_flocker_control(distribution):
     """
     Enable flocker-control service.
     """
-    return sequence([
-        run_from_args(['systemctl', 'enable', 'flocker-control']),
-        run_from_args(['systemctl', 'start', 'flocker-control']),
-    ])
+    if distribution in ('centos-7', 'fedora-20'):
+        return sequence([
+            run_from_args(['systemctl', 'enable', 'flocker-control']),
+            run_from_args(['systemctl', 'start', 'flocker-control']),
+        ])
+    elif distribution == 'ubuntu-14.04':
+        # Since the flocker-control service is currently installed
+        # alongside the flocker-agent service, the default control
+        # service configuration does not automatically start the
+        # service.  Here, we provide an override file to start it.
+        return sequence([
+            put(
+                path='/etc/init/flocker-control.override',
+                content=dedent('''\
+                    start on runlevel [2345]
+                    stop on runlevel [016]
+                    '''),
+            ),
+            run("echo 'flocker-control-api\t4523/tcp\t\t\t# Flocker Control API port' >> /etc/services"),  # noqa
+            run("echo 'flocker-control-agent\t4524/tcp\t\t\t# Flocker Control Agent port' >> /etc/services"),  # noqa
+            run_from_args(['service', 'flocker-control', 'start']),
+        ])
+    else:
+        raise NotImplementedError()
```

Comment:
No need for now - this doesn't exist elsewhere and will be ripped out for FLOC-1670.